great_expectations/util.py

```diff
@@ -2152,3 +2152,37 @@ def numpy_quantile(
         )
 
     return quantile
+
+
+class NotImport:
+    def __init__(self, message: str):
+        self.__dict__["gx_error_message"] = message
+
+    def __getattr__(self, attr: str) -> Any:
+        raise ModuleNotFoundError(self.__dict__["gx_error_message"])
+
+    def __setattr__(self, key: str, value: Any) -> None:
+        raise ModuleNotFoundError(self.__dict__["gx_error_message"])
+
+    def __call__(self, *args, **kwargs) -> Any:
+        raise ModuleNotFoundError(self.__dict__["gx_error_message"])
+
+    def __str__(self) -> str:
+        return self.__dict__["gx_error_message"]
+
+
+def maybe_import_module(module_name: str) -> Union[ModuleType, NotImport]:
```

Comment:
@billdirks I am wondering if the approach we used elsewhere already can help with higher-level methods (please see `head()` in New Datasources (also present in `MetricsCalculator` -- it was a generalization of the approach you showed me, when I found that we already had that in `great_expectations/util.py`.  Thanks.